lib/easy.c

```diff
@@ -608,8 +610,10 @@ static CURLcode wait_or_timeout(struct Curl_multi *multi, struct events *ev)
     /* get the time stamp to use to figure out how long poll takes */
     before = curlx_tvnow();
 
+    timeout_ms = (ev->ms > (long)INT_MAX) ? INT_MAX : ev->ms;
+
     /* wait for activity or timeout */
-    pollrc = Curl_poll(fds, numfds, (int)ev->ms);
+    pollrc = Curl_poll(fds, numfds, timeout_ms);
```

Comment:
Blind casting to `int` could result in negative values, which produce undesirable effects.